internal/manager/manager_tasks.go

```diff
@@ -375,128 +404,147 @@ type StashBoxBatchTagInput struct {
+					tasks = append(tasks, &stashBoxBatchPerformerTagTask{
+						performer:      performer,
 						box:            box,
 						excludedFields: input.ExcludeFields,
-						taskType:       Performer,
 					})
 				}
 			}
-		} else { //nolint:gocritic
-			// The gocritic linter wants to fold this if-block into the else on the line above.
-			// However, this doesn't really help with readability of the current section. Mark it
-			// as nolint for now. In the future we'd like to rewrite this code by factoring some of
-			// this into separate functions.
-
-			// The user has chosen to tag every item in their database
-			if err := s.Repository.WithTxn(ctx, func(ctx context.Context) error {
-				performerQuery := s.Repository.Performer
-				var performers []*models.Performer
-				var err error
-
-				if input.Refresh {
-					performers, err = performerQuery.FindByStashIDStatus(ctx, true, box.Endpoint)
-				} else {
-					performers, err = performerQuery.FindByStashIDStatus(ctx, false, box.Endpoint)
-				}
+		}
+		return nil
+	})
 
-				if err != nil {
-					return fmt.Errorf("error querying performers: %v", err)
-				}
+	return tasks, err
+}
 
-				for _, performer := range performers {
-					if err := performer.LoadStashIDs(ctx, performerQuery); err != nil {
-						return fmt.Errorf("error loading stash ids for performer %s: %v", performer.Name, err)
-					}
+func (s *Manager) batchTagPerformersByNamesOrStashIds(input StashBoxBatchTagInput, box *models.StashBox) []Task {
+	var tasks []Task
 
-					tasks = append(tasks, StashBoxBatchTagTask{
-						performer:      performer,
-						refresh:        input.Refresh,
-						box:            box,
-						excludedFields: input.ExcludeFields,
-						taskType:       Performer,
-					})
-				}
-				return nil
-			}); err != nil {
-				return err
+	for i := range input.StashIDs {
+		stashID := input.StashIDs[i]
+		if len(stashID) > 0 {
+			tasks = append(tasks, &stashBoxBatchPerformerTagTask{
+				stashID:        &stashID,
+				box:            box,
+				excludedFields: input.ExcludeFields,
+			})
+		}
+	}
+
+	names := input.Names
+	if len(names) == 0 {
+		names = input.PerformerNames //nolint:staticcheck
+	}
+
+	for i := range names {
+		name := names[i]
+		if len(name) > 0 {
+			tasks = append(tasks, &stashBoxBatchPerformerTagTask{
+				name:           &name,
+				box:            box,
+				excludedFields: input.ExcludeFields,
+			})
+		}
+	}
+
+	return tasks
+}
+
+func (s *Manager) batchTagAllPerformers(ctx context.Context, input StashBoxBatchTagInput, box *models.StashBox) ([]Task, error) {
+	var tasks []Task
+
+	err := s.Repository.WithTxn(ctx, func(ctx context.Context) error {
+		performerQuery := s.Repository.Performer
+		var performers []*models.Performer
+		var err error
+
+		if input.Refresh {
+			performers, err = performerQuery.FindByStashIDStatus(ctx, true, box.Endpoint)
+		} else {
+			performers, err = performerQuery.FindByStashIDStatus(ctx, false, box.Endpoint)
+		}
```

Comment:
This can be simplified to `...FindByStashIDStatus(ctx, input.Refresh, box.Endpoint)`